python/sglang/srt/managers/data_parallel_meta.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import multiprocessing as mp
+import pickle
+import struct
+from multiprocessing import shared_memory
+from multiprocessing.managers import BaseManager
+from typing import Dict, List
+
+logger = logging.getLogger(__name__)
+
+"""
+This class will be use in scheduler and dp controller
+If this class is placed in the dp controller,
+it will cause circular references, so it is placed in a separate file.
+"""
+
+
+class DPBalanceMeta:
+    def __init__(self, num_workers: int):
+        self.num_workers = num_workers
+        self._manager = mp.Manager()
+        self.mutex = self._manager.Lock()
```

Comment:
I was thinking if we could abstract some methods to `python/sglang/srt/utils.py` and `python/sglang/srt/distributed/parallel_state.py` ?